pipeline_utils/pipeline_deploy.py

```diff
@@ -0,0 +1,406 @@
+        }
+        self.filepath = {
+            'Software': 'portal_objects/software.yaml',
+            'FileFormat': 'portal_objects/file_format.yaml',
+            'FileReference': 'portal_objects/file_reference.yaml',
+            'Workflow': 'portal_objects/workflows',
+            'MetaWorkflow': 'portal_objects/metaworkflows',
+            'ECR': 'dockerfiles',
+            'WFL': 'wfl'
+        }
+
+        # Load attributes
+        for key, val in vars(args).items():
+            setattr(self, key, val)
+
+        # Get pipeline version
+        with open(f'{self.repo}/{version}') as f:
+            self.version = f.readlines()[0].strip()
+        # Get pipeline name
+        with open(f'{self.repo}/{pipeline}') as f:
+            self.pipeline = f.readlines()[0].strip()
+
+        # Load credentials
+        # self._get_credentials()
+
+
+    def _get_credentials(self):
+        """Get auth credentials to target portal and environment.
+        """
+        # Get portal credentials
+        if os.environ.get('GLOBAL_BUCKET_ENV') and os.environ.get('S3_ENCRYPT_KEY'):
+            # new account
+            s3 = s3_utils.s3Utils(env=self.ff_env)
+            self.ff_key = s3.get_access_keys('access_key_admin')
+        elif os.environ.get('S3_ENCRYPT_KEY'):
+            # main account, also need the key to get auth this way
+            self.ff_key = ff_utils.get_authentication_with_server(ff_env=self.ff_env)
+        else:
+            # read key for portal auth from self.keydicts_json [~/.cgap-keydicts.json]
+            try:
+                with open(os.path.expanduser(self.keydicts_json)) as keyfile:
+                    keys = json.load(keyfile)
+            except Exception:
+                error = 'ERROR, could not locate file with key dicts for portal auth --keydicts-json\n'
+                sys.exit(error)
+            self.ff_key = keys.get(self.ff_env)
+
+        # Get encryption key
+        self.kms_key_id = os.environ.get('S3_ENCRYPT_KEY_ID', None)
+
+        # Check args
+        if not self.ff_key:
+            error = 'ERROR, missing key for {0} environment in key dicts for auth\n'.format(self.ff_env)
+            sys.exit(error)
+
+
+    def _post_patch_json(self, data_json, type):
+        """Routine to POST|PATCH metadata from JSON object to portal.
+        """
+        if not self.debug:
+            is_patch = True
+            # try:
+            #     ff_utils.get_metadata(data_json['aliases'][0], key=self.ff_key)
+            # except Exception:
+            #     is_patch = False
+
+            # Exception for uploading of FileReference objects
+            #   status -> uploading, uploaded
+            #   default is None -> the status will not be updated during patch,
+            #     and set to uploading if post for the first time
+            if type == 'FileReference':
+                # main status
+                if data_json['status'] == None:
```

Comment:
Consider to handle linting error
```suggestion
                if data_json['status'] is None:
```

Reply:
done